### PR TITLE
Fix/update leaderboards docs

### PR DESCRIPTION
**Pull request by FractalJackal**
This PR includes commits to various bits of the Leaderboards docs to update the service name from "Momento Leaderboard" to "Momento Leaderboards". I also removed mention of it being in beta from the index page. The only thing I wasn't sure how to suggest a change on is the name of the folder for this part of the docs (updating from "leaderboard" to "leaderboards").

Please review and let me know any feedback on how I did this. Thanks!

### CODE REVIEW

**Review by eaddingtonwhite — DISMISSED** [review]
woohoo go @FractalJackal ! 🐐

**Comment by anitarua** [comment]
Are you trying to edit the sidebar? https://github.com/momentohq/public-dev-docs/blob/main/sidebars.js

Or maybe it's this: https://github.com/momentohq/public-dev-docs/blob/main/docusaurus.config.js#L286

I vaguely remember touching these files for something like that

**Comment by FractalJackal** [comment]
@anitarua Yes, I think maybe both of those are what I'm looking for. @eaddingtonwhite Do you know if I update the leaderboard mentions in those to leaderboards if that will work?

**Comment by eaddingtonwhite** [comment]
Yeah i think @anitarua got right links @FractalJackal . Can also show you how to run this locally as well if want so can see changes locally first.

**Review by mjindalmotiv8 — DISMISSED** [review]
Thank you

File: docs/leaderboard/index.md [hunk 1]
```diff
@@ -57,11 +57,11 @@ Check out the [Node.js example](https://github.com/momentohq/client-sdk-javascri
...
+  Momento Leaderboards has a much more efficient memory footprint, allowing us to scale to tens of millions of records, and the APIs were built with gaming use cases specifically in mind.
 </details>
 
 <details>
-  <summary>Do Momento Leaderboard items expire?</summary>
-  No, Momento Leaderboard items have no TTL.
-</details>
\ No newline at end of file
+  <summary>Do Momento Leaderboards items expire?</summary>
+  No, Momento Leaderboards items have no TTL.
```
Comment (eaddingtonwhite) [inline docs/leaderboard/index.md]: should this answer be updated to say 7 day ttl by default?

Comment (anitarua) [reply docs/leaderboard/index.md]: haha yes probably

**Comment by anitarua** [comment]
Screenshot 2023-10-18 at 1 02 34 PM

**Review by allenheltondev — APPROVED** [review]

**Pull request merged**